### PR TITLE
Making a start on isolating the unit tests from the geocoding service

### PR DESCRIPTION
**Pull request by tansaku**
This isn't complete, but is a quick and dirty stubbing of a test that was hitting the geocoding service through the use of factory girl.

Basically we either need to drop factory girl for organisations, or adjust the way we use validations (or both).

See http://samuelmullen.com/2013/05/the-problem-with-rails-callbacks/

I think the key problem is that currently various controller tests are hitting a good 3/4 of the stack, which our acceptance tests are already doing, and are about to be doing in a much nicer sandboxed fashion.

If we have 'request' specs that are testing our endpoints in ways that are difficult to do with acceptance tests, that's fine, but they need to be separated from any "unit tests" and have proper VCR coverage.

If they are "unit tests" then they shouldn't be testing private methods, and they shouldn't be testing multiple elements of the stack as they end up being too closely tied to our implementation.  Proper unit testing should help us create SOLID objects and avoid convoluted pathways of logic that are hard to follow.

### CODE REVIEW

**Comment by tansaku** [comment]
anyhow the key thing here is that this is the start of a quick fix to isolate these tests from 3rd party APIs, and then there is a bigger process of refactoring required to make these tests really drive quality in our code base ...

**Comment by johnnymo87** [comment]
I can understand what you're doing here, but I feel like mocking controller tests is only the right solution *after* refactoring all the business logic out of the controller. Otherwise, we're left with tests that make it extremely difficult to ever refactor. (@mtc2013 and I felt this pain when pairing on the gmaps upgrade a year and a half ago: https://github.com/AgileVentures/LocalSupport/pull/115)


What you've been doing with the service objects in other PRs seems like a good solution. Let's say the index method read like this (pardon my un-DDD object name):
```
  def index
    reader = VolunteerOps::Read.new
    @volunteer_ops = reader.volunteer_ops
    @organisations = reader.organisations
    @markers = reader.markers
  end
```
I think if our controller code read like this, we'd be in a very good place to write mock controller tests, since the only thing happening in our controllers is the assignment of variables. All business logic would reside behind the veil of the service object, and could be tested separately there (with the use of the `FakeGoogleGeocode` or whatever tool we're currently using to isolate ourselves from 3rd party APIs).

If you'd like, I'll open a PR with this kind of refactor?

**Comment by tansaku** [comment]
oh definitely - we totally want to more to this kind of service object usage.  Would love to see a PR like that from you.  

As I was saying - I'm focusing on a chore that is sandboxing every network connection and clearing the cruft out of the rspec and cucumber logs, and trying not to get sidetracked from that.

I don't know that switching from FactoryGirl to mock_model makes things harder to refactor thought.  Anyway, either way I'd love to see your pull request.  A vision of how to make all these things makes sense is forming I think :-)

**Comment by tansaku** [comment]
aha and with this latest commit we have a completely clean rspec log - woot! and it makes it look like maybe I could have got away with an even simpler change on the ops controller to remove the gecoding hit - maybe ...?

**Comment by tansaku** [comment]
@mtc2013 @ZmagoD @decareano @marianmosley - your thoughts on this much appreciated.  Any reason I shouldn't just merge this in in order to clear out the rspec log?

@johnnymo87 I know what you're saying about making things more difficult to refactor, but isn't sandboxing hitting 3rd party network services a higher priority?

I think that the high level testing is the job of the acceptance tests (which allow lots of unit level refactoring).  Here I'm basically replacing FactoryGirl with mock_model to avoid triggering the lifecycle hooks that lead to a network interaction.

I'm not a big fan of mock_model, but I'm no fan of FactoryGirl either.  In the long run I'd like to make everything about testing Plain old ruby objects, in services as you suggest; but that each service should be individually tested at the unit level, and their integration tested by acceptance tests if at all possible ... at least that's what my Makers experience is leading me towards ...
